03/Product.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+function Product(props) {
+    const {isCategory, isCart, product, isProductPurchased, addToCart, removeFromCart} = props
+    let productContent = null
+
+    if(isCategory) {
+        productContent = <button disabled={ isProductPurchased(product.id) } onClick={ () => addToCart(product) } >Dodaj do koszyka</button>
+    }
+
+    if(isCart) {
+        productContent = <button onClick={ () => removeFromCart(product) } >Usuń z koszyka</button>
+    }
```

Comment:
Myślę, że lepiej byłoby przekazywać w `props`, treść buttona oraz odpowiednią funkcję.
Wtedy kod wew. tego komponentu zdecydowanie byłby prostszy i uniwersalny.
Spróbuj to zmienić ;)